clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -8656,6 +8656,9 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
       // lowered to CrossWorkgroup storage class that is mapped to just
       // global address space.
       ExtArg += ",+SPV_INTEL_usm_storage_classes";
+    else
+      // Don't enable several freshly added extensions on FPGA H/W
+      ExtArg += ",+SPV_INTEL_token_type";
```

Comment:
Unless I'm missing some other condition, negation of `!isFPGAEmulationMode()` (line ) is only going to enable the extension for FPGA Emulation and bypass all other targets. The LITs are positive, though, so how exactly does this work out?

Reply:
AFAIK `FPGAEmulationMode = true` by default and only FPGA H/W setting is changing it to `false`. I believe it's better at some point reverse `FPGAEmulationMode = true` to `FPGAHardwareMode = false` , so the code like this wouldn't confuse anybody.

Reply:
Thanks, fair point. Maybe a `enum` of some sort would also be an option (to dodge any bugs in FPGA-irrelevant driver code). Could you please file a GH issue for that?

Reply:
Done in https://github.com/intel/llvm/issues/4061